my-sms/spec/controllers/students_controller_spec.rb

```diff
@@ -23,119 +25,114 @@
-        post :create, {:student => invalid_attributes}, valid_session
-        expect(response).to be_success
+        expect(subject).to be_success
       end
     end
   end
 
-  describe "PUT #update" do
-    context "with valid params" do
-      let(:new_attributes) {
-        skip("Add a hash of attributes valid for your model")
-      }
-
-      it "updates the requested student" do
-        student = Student.create! valid_attributes
-        put :update, {:id => student.to_param, :student => new_attributes}, valid_session
-        student.reload
-        skip("Add assertions for updated state")
+  describe 'PUT #update' do
+    let(:new_student_args) { {} }
+    subject do
+      student = FactoryBot.create(:student)
+      new_attributes = FactoryBot.attributes_for(:student, **new_student_args)
+      put :update, { id: student.to_param, student: new_attributes }
+    end
+
+    context 'with valid params' do
+      it 'updates the requested student' do
+        subject do
+          student.reload
+          attrs = %i[first_name last_name gender]
+          student_attrs = student.attributes.fetch_values(*attrs.map(&:to_s))
+          new_attrs = new_attributes.fetch_values(*attrs)
+          expect(student_attrs).to eq new_attrs
+        end
       end
 
-      it "redirects to the student" do
-        student = Student.create! valid_attributes
-        put :update, {:id => student.to_param, :student => valid_attributes}, valid_session
-        expect(response).to redirect_to(student)
+      it 'redirects to the student' do
+        subject do
+          expect(response).to redirect_to(student)
+        end
       end
     end
 
-    context "with invalid params" do
+    context 'with invalid params' do
+      let(:new_student_args) { { first_name: nil } }
+
       it "returns a success response (i.e. to display the 'edit' template)" do
-        student = Student.create! valid_attributes
-        put :update, {:id => student.to_param, :student => invalid_attributes}, valid_session
-        expect(response).to be_success
+        expect(subject).to be_success
       end
     end
   end
 
-  describe "DELETE #destroy" do
-    it "destroys the requested student" do
-      student = Student.create! valid_attributes
-      expect {
-        delete :destroy, {:id => student.to_param}, valid_session
-      }.to change(Student, :count).by(-1)
+  describe 'DELETE #destroy' do
+    subject do
+      student = FactoryBot.create(:student)
+      delete :destroy, { id: student.to_param }
     end
 
-    it "redirects to the students list" do
-      student = Student.create! valid_attributes
-      delete :destroy, {:id => student.to_param}, valid_session
-      expect(response).to redirect_to(students_url)
+    it 'destroys the requested student' do
+      expect { subject }.to change(Student, :count).by(0)
     end
-  end
 
+    it 'redirects to the students list' do
+      expect(subject).to redirect_to(students_url)
+    end
+  end
```

Comment:
This is a good example of `subject` here, but the `student = ...` should be moved to a let binding instead

Reply:
I've fixed this pattern in 551e65cb9e69cc4946a34aa25e79fee7598465dd